### PR TITLE
Accept FQDN for whitelist

### PR DESCRIPTION
**Pull request by stephdl**
We can accept FQDN/Domain name for whitelisting, when you do not have a static IP it can help you to not be locked out.

https://github.com/NethServer/dev/issues/6491

### CODE REVIEW

File: api/validate [hunk 1]
```diff
@@ -43,15 +43,21 @@ switch ($data['action']) {
...
+            } elseif ($ipCidrAllow->evaluate($el)) {
+                continue;
+            } elseif ($FQDNAllow->evaluate($el)){
+                continue;
+            } else {
+                $v->addValidationError('IgnoreIP', 'not_a_valid_IP_or_CIDR_or_FQDN', $el);
             }
         }
 
+
```
Comment (gsanchietti) [inline api/validate:60]:
```suggestion
```

File: ui/public/i18n/language.json [hunk 2]
```diff
@@ -117,7 +117,7 @@
     },
     "validation":{
         "validation_failed": "Validation failed",
-        "not_a_valid_IP_or_CIDR": "It must be a IPv4 address like '1.2.3.4' or a CIDR network like '192.168.1.0/24'",
+        "not_a_valid_IP_or_CIDR_or_FQDN": "It must be a IPv4 address like '1.2.3.4', a CIDR network like '192.168.1.0/24' or a domain name like 'my.domain.com'",
```
Comment (gsanchietti) [inline ui/public/i18n/language.json:120]:
```suggestion
        "not_a_valid_IP_or_CIDR_or_FQDN": "It must be a IPv4 address like '1.2.3.4', a CIDR network like '192.168.1.0/24' or a fully qualified domain name like 'my.domain.com'",
```

File: ui/public/i18n/language.json [hunk 3]
```diff
@@ -128,7 +128,7 @@
     "docs":{
         "fail2ban": "Fail2ban",
         "more_info_about": "More Information about ",
-        "IgnoreIP": "All IP (1.2.3.4) or CIDR network (192.168.1.0/24) in this list will never be banned by Fail2ban. Add one entry per line",
+        "IgnoreIP": "All IP (1.2.3.4), CIDR network (192.168.1.0/24) or domain name (my.domain.com) in this list will never be banned by Fail2ban. Add one entry per line",
```
Comment (gsanchietti) [inline ui/public/i18n/language.json:131]:
```suggestion
        "IgnoreIP": "All IP (1.2.3.4), CIDR network (192.168.1.0/24) or a fully qualified domain name (my.domain.com) in this list will never be banned by Fail2ban. Add one entry per line",
```

**Review by gsanchietti — COMMENTED** [review]
The PR good, I'd like to suggest only a couple of cosmetic fixes.

**Comment by stephdl** [comment]
@gsanchietti I validate domain and FQDN, both are true, do you think only FQDN is better or it is the same ?

**Comment by gsanchietti** [comment]
> @gsanchietti I validate domain and FQDN, both are true, do you think only FQDN is better or it is the same ?

I'm not sure. I saw you used `HOSTNAME_FQDN` validator, so I assumed only a FQDN was valid.
So, `my` is not valid, but both `domain.org` and `my.domain.org` are valid. Seem good to me.

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**